bindings/python/scripts/parse.py

```diff
@@ -1,145 +1,89 @@
-
+        if node_in_this_file(node=child, filename=this_filename):
+            walk_and_print(cursor=child, this_filename=this_filename, depth=depth + 1)
 
-def dump_json(filepath, parsed_list):
-    with open(filepath, "w") as f:
-        json.dump(parsed_list, f, indent=2)
 
+def generate_parsed_info(cursor, this_filename, depth):
+    parsed_info = dict()
 
-def generate_parsed_info(cursor, filter, lines, this_filename, depth, parsed_list):
     if cursor.spelling:
-        holder = {
-            "depth": depth,
-            "line": cursor.location.line,
-            "column": cursor.location.column,
-            "kind": cursor.kind.name,
-            "name": cursor.spelling,
-        }
+        parsed_info["depth"] = depth
+        parsed_info["line"] = cursor.location.line
+        parsed_info["column"] = cursor.location.column
+        parsed_info["kind"] = cursor.kind.name
+        parsed_info["name"] = cursor.spelling
         if cursor.type.kind.spelling != "Invalid":
-            holder["element_type"] = cursor.type.kind.spelling
+            parsed_info["element_type"] = cursor.type.kind.spelling
         if cursor.access_specifier.name != "INVALID":
-            holder["access_specifier"] = cursor.access_specifier.name
+            parsed_info["access_specifier"] = cursor.access_specifier.name
         if cursor.result_type.spelling != "":
-            holder["result_type"] = cursor.result_type.spelling
+            parsed_info["result_type"] = cursor.result_type.spelling
         if cursor.brief_comment:
-            holder["brief_comment"] = cursor.brief_comment
+            parsed_info["brief_comment"] = cursor.brief_comment
         if cursor.raw_comment:
-            holder["raw_comment"] = cursor.raw_comment
-        holder["members"] = []
-
-        parsed_list.append(holder)
+            parsed_info["raw_comment"] = cursor.raw_comment
+        parsed_info["members"] = []
 
     for child in cursor.get_children():
-        if node_in_this_file(child, this_filename):
-            child_list = []
-            generate_parsed_info(
-                child, filter, lines, this_filename, depth + 1, child_list,
-            )
-            if child_list and parsed_list:
-                if len(parsed_list[0]["members"]):
-                    parsed_list[0]["members"].append(child_list[0])
-                else:
-                    parsed_list[0]["members"] = child_list
+        if node_in_this_file(node=child, filename=this_filename):
+            child_parsed_info = generate_parsed_info(cursor=child, this_filename=this_filename, depth=depth + 1,)
+            if child_parsed_info and parsed_info:
+                parsed_info["members"].append(child_parsed_info)
 
+    return parsed_info
 
-def parse_arguments(args):
-    parser = argparse.ArgumentParser(description="C++ libclang parser")
-    parser.add_argument("files", nargs="+", help="The source files to search")
-    return parser.parse_args(args)
 
+def create_index():
```

Comment:
I can remove it if we don't want to test it. I modified the main function so that it calls either the file's(`parse.py`) functions or `utils.py`'s functions so that everything the main function does, can be tested.

Reply:
Since this is only a call to an external function, you'd be testing their implementation, not yours. It wouldn't be useful to prevent your mistakes, only theirs

Reply:
Understood. Removed.